yugabyte/terraform-aws-yugabyte
# Added the Terratest plan and Github action (CI) to deploy and test on AWS.

Co-authored-by: Sameer Kulkarni <samkulkarni20@gmail.com>
Co-authored-by: Yatish Sharma <yatish1996ss@gmail.com>
Signed-off-by: Yatish Sharma <yatish1996ss@gmail.com>

- Added provider module in main.tf.
- Push event will trigger the CI workflow.
- Introduced a new variable called `allowed_sources` to restrict the traffic on the security group so only the mentioned sources can communicate with the cluster.
- To verify the below-mentioned scenarios, it uses Public IP of instances for SSH and Private IP to check APIs.
- Given the meaningful name to Terraform outputs.

The Terratest will test the following scenarios - 
  - Check the status code for the Master and Tserver UI.
  - Verify the SSH connectivity.
  - Verify the YSQL and YCQL APIs.
  - Check Master and Tserver conf exists or not, and If exists, then print the conf.
  - Check the Tserver Process status.
  - Check the Tserver logfile exists or not.
  - Check the Ulimits configuration of the node.

This PR is a refined version of this PR: https://github.com/yugabyte/terraform-aws-yugabyte/pull/23
